### PR TITLE
alternative ascii-int program

### PR DESCRIPTION
**Pull request by cnorthway**
alternative to current implementation.
- uses integers and writes integer-sized output (we need to handle values ~20,000)
- converts one int at a time (won't get out of sync, might be able to be reused elsewhere)
- error messages
- creates the input fifo (not necessarily better but lines up with other programs)
- relies on library functions to do the parsing & conversion

### CODE REVIEW

File: convert.c [hunk 1]
```diff
@@ -1,67 +0,0 @@
...
-int main(int ac, char *av[])
-{
-    assert(ac == 3 && "usage: ./convert in-fifo out-fifo");
-    int result = 0, val1 = 0, val2 = 0, buffset = 0, offset1 = 0, offset2 = 0;
-    char *buffer;
-    char arr1[100];
-    char arr2[100];
-
-    size_t size = 32;
-    u_int8_t output[2];
```
Comment (cnorthway) [inline convert.c:17]: uint8_t is too small for our values

File: Makefile [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 FLAGS = -g -Wall -Werror -std=c99
 
 clean:
-	rm convert
+	rm ascii_conv
 
-conver: convert.c
-	gcc $(FLAGS) -o $@ $^
\ No newline at end of file
+ascii_conv: ascii_conv.c
```
Comment (jleightcap) [inline Makefile:8]: have the convention that are binary names have hyphens !!!

File: ascii_conv.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  if (out_fd == -1) {
+    syslog(LOG_ERR, "opening bin output fifo failed (%m)");
+  }
+  while (1) {
+    int val;
+    int ret = fscanf(in_file, "%i", &val);
+    if (ret == EOF) {
+      if (ferror(in_file) != 0) {
+        // not end of file (expected), we would like to know what's goin on
+        syslog(LOG_INFO, "ascii integer read error");
```
Comment (jleightcap) [inline ascii_conv.c:29]: did you ever trigger this on testing? like, assuming we set up the pipelines right, this should be unreachable correct?

File: testing.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+### expected behavior:
+
+converts whitespace-separated ascii representations of integers into raw
+integers (C's `int` type, signed, 4 bytes)
```
Comment (jleightcap) [inline testing.md:4]: platform dependent, this is not strictly true :-)

**Review by jleightcap — APPROVED** [review]

**Review by nfantasia — APPROVED** [review]

File: ascii_conv.c (hunk 3 above)
Comment (cnorthway) [reply ascii_conv.c:29]: as long as the pipe doesn't die this shouldn't be an issue

**Pull request merged**